distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -283,3 +102,169 @@ type Marshalling_Error
+                text : Text ->
+                    set_value current text (pair.at 1).to_js_object
+                    current
+                _ -> Error.throw (Illegal_Argument.Error "JS_Object.from_pairs: key must be a Text value")
+        JS_Object.Value js_object
+
+    ## PRIVATE
+    Value js_object
+
+    ## Get a value for a key of the object, or a default value if that key is not present.
+
+       Arguments:
+       - key: The key to get.
+       - if_missing: The value to return if the key is not found.
+    get : Text -> Any -> JS_Object | Boolean | Number | Nothing | Text | Vector
+    get self key ~if_missing=Nothing =
+        if (has_property self.js_object key) . not then if_missing else
+            value = get_value self.js_object key
+            make_enso value
+
+    ## Get a value for a key of the object.
+       If the key is not found, throws a `No_Such_Key` error.
+
+       Arguments:
+       - key: The key to get.
+    at : Text -> JS_Object | Boolean | Number | Nothing | Text | Vector ! No_Such_Key
+    at self key = self.get key (Error.throw (No_Such_Key.Error self key))
+
+    ## Get the keys of the object.
+    field_names : Vector
+    field_names self =
+        Vector.from_polyglot_array (get_property_names self.js_object)
+
+    ## Gets the number of keys in the object.
+    length : Number
+    length self =
+        get_property_names self.js_object . length
+
+    ## Convert the object to a Vector of Pairs.
+    to_vector : Vector
+    to_vector self =
+        keys = get_property_names self.js_object
+        proxy = Array_Proxy.new keys.length (i-> Pair.new (keys.at i) (self.get (keys.at i)))
+        Vector.from_polyglot_array proxy
+
+    ## Convert the object to a JS_Object.
+    to_js_object : JS_Object
+    to_js_object self = self
+
+    ## Convert to a Text.
+    to_text : Text
+    to_text self = Json.stringify self
+
+    ## Convert to a JSON representation.
+    to_json : Text
+    to_json self = self.to_text
+
+    ## Checks if this JS_Object is equal to another JS_Object.
+
+       Arguments:
+       - that: The map to compare `self` to.
+    == : JS_Object -> Boolean
+    == self that = case that of
+        _ : JS_Object ->
+            self_keys = self.field_names
+            that_keys = that.field_names
+            self_keys.length == that_keys.length && self_keys.all key->
+                (self.get key == that.at key).catch No_Such_Key.Error _->False
+        _ -> False
+
+    ## UNSTABLE
+
+       Transform the vector into text for displaying as part of its default
+       visualization.
+    to_default_visualization_data : Text
+    to_default_visualization_data self =
+        render self
+
+## PRIVATE
+   Render the JS_Object to Text with truncated depth.
+render object depth=0 max_depth=5 = case object of
+    _ : JS_Object ->
+        if depth == max_depth then '"{...}"' else
+            len = object.length
+            keys = object.field_names.take 50
```

Comment:
Sorry for another nitpick, feel free to ignore it:
While we do not have Enso style guide yet, IIRC most of our style guides for other languages forbid magic numbers. Maybe we can assign this to a variable and give it a name? It will be more meaningful and if it ever needs changing we need to change just one place.